vscode-dotnet-runtime-library/src/Acquisition/LinuxVersionResolver.ts

```diff
@@ -0,0 +1,288 @@
+                throw error;
+            }
+
+            const pair : DistroVersionPair = { distro : distroName, version : distroVersion };
+            return pair;
+        }
+        catch(error)
+        {
+            const err = new DotnetAcquisitionDistroUnknownError(new Error(`${this.baseUnsupportedDistroErrorMessage} ... does /etc/os-release exist?`));
+            this.acquisitionContext.eventStream.post(err);
+            throw err;
+        }
+    }
+
+
+    /**
+     * @remarks we need to call the command executor to allow us to test this class.
+     * The command executor needs to be able to run async commands. We cant call async commands in the constructor.
+     * So we have to use this pattern of calling initialize to verify our members are not null in each function.
+     */
+    public async Initialize()
+    {
+        if(!this.distro)
+        {
+            this.distro = await this.getRunningDistro();
+        }
+
+        if(!this.distroSDKProvider)
+        {
+            this.distroSDKProvider = this.DistroProviderFactory(this.distro);
+        }
+
+        if(!this.distro || !this.distroSDKProvider)
+        {
+            const error = new DotnetAcquisitionDistroUnknownError(new Error(`${this.baseUnsupportedDistroErrorMessage} ... we cannot initialize.`));
+            this.acquisitionContext.eventStream.post(error);
+            throw error;
+        }
+    }
+
+    private DistroProviderFactory(distroAndVersion : DistroVersionPair) : IDistroDotnetSDKProvider
+    {
+        switch(distroAndVersion)
+        {
+            // Implement any custom logic for a Distro Class in a new DistroSDKProvider and add it to the factory here.
+            case null:
+                const error = new DotnetAcquisitionDistroUnknownError(new Error(this.baseUnsupportedDistroErrorMessage));
+                this.acquisitionContext.eventStream.post(error);
+                throw error;
+            default:
+                return new GenericDistroSDKProvider(distroAndVersion);
+        }
+    }
+
+    /**
+     *
+     * @param supportStatus The support status of this distro and version pair.
+     * @param fullySpecifiedDotnetVersion The version of dotnet requested to install, upgrade, etc.
+     * @remarks Throws a specific error below if a conflicting install type of dotnet exists on linux.
+     * Microsoft and distro feed packages together cause system instability with dotnet, so we dont want to let people get into those states.
+     * Eventually, we could add logic to remove them for users, but that may require consent first.
+     */
+    public async VerifyNoConflictInstallTypeExists(supportStatus : DotnetDistroSupportStatus, fullySpecifiedDotnetVersion : string) : Promise<void>
+    {
+        await this.Initialize();
+
+        if(supportStatus === DotnetDistroSupportStatus.Distro)
+        {
+            const microsoftFeedDir = await this.distroSDKProvider!.getExpectedDotnetMicrosoftFeedInstallationDirectory();
```

Comment:
This feels backwards to me, but maybe I don't have enough context- if the support status is distro, we use the msft feed? And if the support status is msft we use the distro feed?

Reply:
This is in `VerifyNoConflictInstallTypeExists` --> dotnet will fail spectacularly and your machine will be in a broken state if you have both the distro feed and microsoft feed dotnet installed at the same time. So, if it's distro feed support, we check if a microsoft feed install exists and fail if so, and vice-versa ;)